docs/messaging.rst

```diff
@@ -133,6 +133,27 @@ A message is defined by the following four-dictionary structure::
     so implementers are strongly encouraged to include it.
     It will be mandatory in 5.1.
 
+.. _msging_compatibility:
+
+Compatibility
+=============
+
+Kernels must implement the :ref:`execute <execute>` and :ref:`kernel info
+<msging_kernel_info>` messages in order to be usable. All other message types
+are optional, although we recommend implementing :ref:`completion
+<msging_completion>` if possible. Kernels do not need to send any reply for
+messages they don't handle, and frontends should provide sensible behaviour if
+no reply arrives (except for the required execution and kernel info messages).
+
+:ref:`stdin messages <stdin_messages>` are unique in that the request comes from
+the kernel, and the reply from the frontend. The frontend is not required to
+support this, but if it does not, it must set ``'allow_stdin' : False`` in
```

Comment:
(this question is more about an extension of this idea, but definitely not blocking this PR)